sunpy/map/mapbase.py

```diff
@@ -2730,8 +2730,9 @@ def contour(self, level, **kwargs):
             # find_contours expects a scalar below
             level = level[0]
 
-        contours = measure.find_contours(self.data, level=level, **kwargs)
-        contours = [self.wcs.array_index_to_world(c[:, 0], c[:, 1]) for c in contours]
+        gen = contour_generator(z=self.data,name="threaded", **kwargs)
```

Comment:
We need to be intentional about enabling threaded, if we do this we are no longer using the *exact* same code path as mpl.

Reply:
Would you prefer we add an explicit method keyword to choose?

Reply:
no, it's more that we need to decide how important to us algorithm compatibly is to us.

Reply:
Well I mean, why not let users pick, between MPL, scikit-image and contourpy methods?

Reply:
I am less convinced by keeping skimage as an option, but for contourpy it's more that we need to pick the default.

Reply:
Fair, personally I lean towards playing it safe, setting name to be the simple non-threaded approach and then if MPL change the default to the threaded one, we can do the same.